src/js/common/components/form-elements/GrowableTable.jsx

```diff
@@ -52,21 +52,29 @@ class GrowableTable extends React.Component {
     this.setState({ [blankRowData.key]: 'incomplete' });
   }
 
+  findIncomplete() {
+    return _.findKey(this.state, v => v === 'incomplete');
+  }
+
   handleAdd() {
+    // Save existing
+    this.handleSave();
+
     if (this.props.isValidSection(this.props.path, this.props.data)) {
       this.createNewElement();
     }
   }
 
   handleEdit(event) {
-    this.setState({ [event.target.dataset.key]: 'incomplete' });
+    this.setState({ [event.target.dataset.key]: 'edit' });
   }
 
   handleSave(event) {
     this.props.initializeCurrentElement();
+    const key = event ? event.target.dataset.key : this.findIncomplete();
 
     if (this.props.isValidSection(this.props.path, this.props.data)) {
-      this.setState({ [event.target.dataset.key]: 'complete' });
+      this.setState({ [key]: 'complete' });
```

Comment:
One odd side effect of this is that the Update button doesn't work if your second row isn't valid